py/asmrv32.c

```diff
@@ -126,7 +100,6 @@ static void split_immediate(mp_int_t immediate, mp_uint_t *upper, mp_uint_t *low
     // Turn the lower half from unsigned to signed.
     if ((*lower & 0x800) != 0) {
         *upper += 0x1000;
-        *lower -= 0x1000;
```

Comment:
Is removing this line intentional?  Is it a bug fix or refactoring?

Reply:
More like both a bug fix and a refactoring.  `*lower` would contain the lowest 12 bits of a 32 bits immediate, so applying a further offset isn't correct in that case even if it leaves the lowest part alone; however the variable has its upper 20 bits masked out on every usage so the extra offset didn't really matter.

Reply:
OK